Dikubot/Webapp/Data/FacebookService.cs

```diff
@@ -150,14 +159,39 @@ public List<FacebookPageModel> AddLongLivedAccessTokenToDatabase(List<FBPage> fa
         foreach (var facebookPageItem in facebookPagesList)
         {
             FacebookPageModel facebookPageModel = new FacebookPageModel();
-            if (_facebookPageService.Exists(m => m.PageId == facebookPageItem.Id))
+            if (FacebookPageService.Exists(m => m.PageId == facebookPageItem.Id))
+            {
+                facebookPageModel = FacebookPageService.Get(m => m.PageId == facebookPageItem.Id);
+            }
+            facebookPageModel.FbPageConverter(facebookPageItem);
+            FacebookPageService.Upsert(facebookPageModel);
+            list.Add(facebookPageModel);
+        }
+        return list;
+    }
+    
+    public List<FacebookPageModel> AddLongLivedAccessTokenToRequestDatabase(List<FBPage> facebookPagesList, ContactModel contact = null)
+    {
+        List<FacebookPageModel> list = new List<FacebookPageModel>();
+        contact ??= new ContactModel(){Name = "NA", Mail = "NA", Number = "NA"};
+        
+        //Add Facebook page to the Databases with Long Lived Access Token
+        foreach (var facebookPageItem in facebookPagesList)
+        {
+            FacebookPageModel facebookPageModel = new FacebookPageModel();
+            if (FacebookPageService.Exists(m => m.PageId == facebookPageItem.Id))
             {
-                facebookPageModel = _facebookPageService.Get(m => m.PageId == facebookPageItem.Id);
+                facebookPageModel = FacebookPageService.Get(m => m.PageId == facebookPageItem.Id);
             }
```

Comment:
You can avoid a db call by refactoring this:
```csharp
FacebookPageModel facebookPageModel = new FacebookPageModel();
if (FacebookPageService.Exists(m => m.PageId == facebookPageItem.Id))
{
     facebookPageModel = FacebookPageService.Get(m => m.PageId == facebookPageItem.Id);
}
```
To:
```csharp
FacebookPageModel facebookPageModel = FacebookPageService.Get(m => m.PageId == facebookPageItem.Id) ?? new FacebookPageModel()
```
But it's not very important